### PR TITLE
Fixes linter issues, upgrades linter

### PR DESCRIPTION
**Pull request by berndverst**
- upgrade to golangci-lint 1.50.1,
- fix linter issues,
- disable deprecated linters

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

### CODE REVIEW

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Can you please update the linter version in a few more places:

- Docs such as these here: https://github.com/dapr/dapr/tree/master/docs/development
- The dev container file

File: .golangci.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ run:
     - ^pkg.*client.*clientset.*versioned.*
     - ^pkg.*client.*informers.*externalversions.*
     - ^pkg.*proto.*
+    - ^pkg.proto.*
```
Comment (berndverst) [inline .golangci.yml:28]: Didn't mean to commit this -- let me remove here.

```suggestion
```

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by mukundansundar — APPROVED** [review]

**Review by daixiang0 — APPROVED** [review]
LGTM

**Comment by daixiang0** [comment]
Please fix CI, as
https://github.com/dapr/dapr/actions/runs/3448087615/jobs/5771538556#step:15:155

**Review by daixiang0 — APPROVED** [review]

**Pull request merged**